ibc/store/proofs_ics23_test.go

```diff
@@ -0,0 +1,178 @@
+package store
+
+import (
+	"crypto/sha256"
+	"testing"
+
+	ics23 "github.com/cosmos/ics23/go"
+	"github.com/pokt-network/pocket/persistence/kvstore"
+	"github.com/pokt-network/smt"
+	"github.com/stretchr/testify/require"
+)
+
+// Proof generation cannot fail but verification can
+func TestICS23Proofs_GenerateCommitmentProofs(t *testing.T) {
+	nodeStore := kvstore.NewMemKVStore()
+	tree := smt.NewSparseMerkleTree(nodeStore, sha256.New(), smt.WithValueHasher(nil))
+	require.NotNil(t, tree)
+
+	// Set a value in the store
+	err := tree.Update([]byte("foo"), []byte("bar"))
+	require.NoError(t, err)
+	err = tree.Update([]byte("bar"), []byte("foo"))
+	require.NoError(t, err)
+	err = tree.Update([]byte("testKey"), []byte("testValue"))
+	require.NoError(t, err)
+	err = tree.Update([]byte("testKey2"), []byte("testValue2"))
+	require.NoError(t, err)
+
+	testCases := []struct {
+		key           []byte
+		value         []byte
+		nonmembership bool
+		fails         bool
+		expected      error
+	}{
+		{
+			// Successfully generates a membership proof for a key stored
```

Comment:
You should name these and use the name below. This way, when a test fails, the name shows up in terminal (or CI)